src/senaite/core/upgrade/v02_04_000.py

```diff
@@ -578,3 +578,44 @@ def purge_backreferences_analysisrequest(tool):
         obj._p_deactivate()
 
     logger.info("Purge stale back-references from samples [DONE]")
+
+
+def migrate_interim_values_to_string(tool):
+    """Migrate all interim values to be string
+    """
+    logger.info("Migrate interim values to string ...")
+
+    uc = api.get_tool("uid_catalog")
+    brains = uc(portal_type=["Analysis", "AnalysisService"])
```

Comment:
include the following as well: `ReferenceAnalysis` and `DuplicateAnalysis`

Reply:
Implemented in b951e6e5c